src/plugins/embeddable/public/lib/editable_embeddable/navigate_to_editor.tsx

```diff
@@ -0,0 +1,71 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import { CoreStart } from '@kbn/core/public';
+import { apiHasParentApi, apiPublishesSearchSession } from '@kbn/presentation-publishing';
+import { firstValueFrom } from 'rxjs';
+import { EmbeddableStart } from '../..';
+import { apiHasAppContext, NavigateToEditorApi } from './types';
+
+/**
+ * Call this function to navigate to a given editor application
+ * @param services The services needed to perform the navigation
+ * @param api The embeddable API, which should, at minimum, provide a callback for fetching
+ *            the editor app target (app ID, path to navigate to, and generic editor URL)
+ */
+export const navigateToEditor = async (
+  services: { core: CoreStart; embeddable: EmbeddableStart },
+  api: NavigateToEditorApi
+) => {
+  const appTarget = await api.getEditorAppTarget();
+  if (appTarget.editApp && appTarget.editPath) {
+    /** The editor has a dedicated app */
+    const apiContext = apiHasAppContext(api) ? api.getAppContext() : undefined;
+    const parentApiContext =
+      apiHasParentApi(api) && apiHasAppContext(api.parentApi)
+        ? api.parentApi.getAppContext()
+        : undefined;
+    let currentAppId = parentApiContext?.currentAppId ?? apiContext?.currentAppId;
+    if (!currentAppId) {
+      /** If the API is not providing a current app ID, then get it from Core */
+      currentAppId = await firstValueFrom(services.core.application.currentAppId$);
+    }
+    const serializedState = api.serializeState?.().rawState;
```

Comment:
Note that `getExplicitInput(embeddable)` is actually quite different from `api.serializeState()` in that `getExplicitInput` is not meant to extract references, but `serializeState` is. Because of this, this line may cause problems with missing references because we simply throw away the references array here. 

Theoretically, if an embeddable's `extract` function modified the state to de-dupe information, that would be lost here. See [here](https://github.com/elastic/kibana/blob/659f373e1fcd81fdf8ad916a67d9b328460d9f2d/examples/embeddable_examples/public/react_embeddables/field_list/field_list_react_embeddable.tsx#L135) for an example of this.

I think this might work for the time being because the legacy embeddables don't de-dupe their references, but in the future maybe we'll pass in the whole `serializedState` object, and modify the apps that do this to call `deserialize` on the state that's passed in.

Could you modify this to check if it's a new or legacy embeddable type and:
- if it's a new type pass in the whole serialized object rather than just the `rawState`? That will force us to handle the extraction / deserialization of this as we convert embeddables.
- If it's an old type, you could call `getExplicitInput` here directly maybe?

Reply:
Good call!! Can do :+1:

Reply:
Done in 7decf57f67c7ab82a3ee1ea35c0537ad20b1f341